src/services/authService.ts

```diff
@@ -0,0 +1,16 @@
+import fetch from '@/utils/request';
+import { AuthCredentials } from '@/constants/types';
+
+const AuthService = async <T>(endpoint: string, credentials: AuthCredentials): Promise<T> => {
+  const { data } = await fetch<T>(`${process.env.NEXT_PUBLIC_API}/users/${endpoint}`, {
```

Comment:
remove the `/users` from the URL and pass it as an argument. 
Change the name to a more generic name because it's not just about auth (requests in the future)